app/controllers/users_controller.rb

```diff
@@ -2,7 +2,14 @@ class UsersController < ApplicationController
   before_action :authenticate_user!
   before_action :only_self_and_admin, except: :show
 
+  def create
+    @user = User.find(params[:user_id])
+    @user.badges.create(badge_params.merge(user: current_user))
```

Comment:
If you say `@user.badges.create` the `badge_params.merge(user: current_user)` is redundant.  It will work fine, but you won't need it.